extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js

```diff
@@ -0,0 +1,235 @@
+/*
+ * VisualEditor user interface WikiaInfoboxInsertDialog class.
+ */
+
+/**
+ * Dialog for inserting portable infobox templates.
+ *
+ * @class
+ * @extends ve.ui.FragmentDialog
+ *
+ * @constructor
+ * @param {Object} [config] Configuration options
+ */
+ve.ui.WikiaInfoboxInsertDialog = function VeUiWikiaInfoboxInsertDialog( config ) {
+	// Parent constructor
+	ve.ui.WikiaInfoboxInsertDialog.super.call( this, config );
+
+	// Properties
+	this.surface = null;
+};
+
+/* Inheritance */
+
+OO.inheritClass( ve.ui.WikiaInfoboxInsertDialog, ve.ui.FragmentDialog );
+
+/* Static Properties */
+
+ve.ui.WikiaInfoboxInsertDialog.static.name = 'wikiaInfoboxInsert';
+
+ve.ui.WikiaInfoboxInsertDialog.static.title = OO.ui.deferMsg( 'wikia-visualeditor-dialog-infobox-insert-title' );
+
+ve.ui.WikiaInfoboxInsertDialog.static.size = 'small';
+
+/* Methods */
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.getSetupProcess = function ( data ) {
+	return ve.ui.WikiaInfoboxInsertDialog.super.prototype.getSetupProcess.call( this, data )
+		.next( function () {
+			this.surface = data.surface;
+		}, this );
+};
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.getBodyHeight = function () {
+	return 600;
+};
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.initialize = function () {
+	// Parent method
+	ve.ui.WikiaInfoboxInsertDialog.super.prototype.initialize.call( this );
+
+	// Initialization
+	this.$content.addClass( 've-ui-wikiaInfoboxInsertDialog' );
+
+	// Load select widget
+	this.loadInfoboxTemplates();
+	this.select = new OO.ui.SelectWidget();
+
+	// Events
+	this.select.connect( this, {
+		select: 'onInfoboxTemplateSelect'
+	} );
+
+	this.$body.append( this.select.$element );
+};
+
+/**
+ * Handle selecting infobox template.
+ *
+ * @method
+ * @param {Object|null} itemData Data of selected item, or null
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.onInfoboxTemplateSelect = function ( itemData ) {
+	var template;
+
+	if ( itemData ) {
+		this.$frame.startThrobbing();
+		this.transclusionModel = new ve.dm.WikiaTransclusionModel();
+		this.transclusionModel.setIsInfobox( true );
+		template = ve.dm.MWTemplateModel.newFromName(
+			this.transclusionModel, itemData.data
+		);
+		this.transclusionModel.addPart( template )
+			.done( this.insertInfoboxTemplate.bind( this ) );
+
+		// TODO: Track
+		ve.track( 'wikia', {
+			action: ve.track.actions.ADD,
+			label: 'infobox-template-insert-from-plain-list'
+		} );
+	}
+};
+
+/**
+ * Prepare infobox template names
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.loadInfoboxTemplates = function () {
+	this.getInfoboxTemplates().done(
+		this.showResults.bind( this )
+	);
+};
+
+/**
+ * Fetch infobox template names from API
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.getInfoboxTemplates = function () {
+	var deferred;
+	if ( !this.gettingTemplateNames ) {
+		deferred = $.Deferred();
+		ve.init.target.constructor.static.apiRequest( {
```

Comment:
is this request being cached on browser side or we are doing it each time the modal is opened?